components/base/wheeled/wheeled_base.go

```diff
@@ -130,64 +130,56 @@ func (wb *wheeledBase) Reconfigure(ctx context.Context, deps resource.Dependenci
 
 	wb.spinSlipFactor = newConf.SpinSlipFactor
 
-	// Check if wb.left is different from newConf.Left before changing wb.left
-	if len(wb.left) != len(newConf.Left) {
-		// Resetting the left motor list
-		wb.left = make([]motor.Motor, 0)
-
-		for _, name := range newConf.Left {
-			m, err := motor.FromDependencies(deps, name)
-			if err != nil {
-				return errors.Wrapf(err, "no left motor named (%s)", name)
+	updateMotors := func(curr *[]motor.Motor, fromConfig *[]string) ([]motor.Motor, error) {
+		newMotors := make([]motor.Motor, 0)
+		if len(*curr) != len(*fromConfig) {
+			for _, name := range *fromConfig {
+				select {
+				case <-ctx.Done():
+					return newMotors, resource.NewBuildTimeoutError(wb.Name())
+				default:
+				}
+				m, err := motor.FromDependencies(deps, name)
+				if err != nil {
+					return newMotors, errors.Wrapf(err, "no left motor named (%s)", name)
+				}
+				newMotors = append(newMotors, m)
 			}
-			wb.left = append(wb.left, m)
-		}
-	} else {
-		// Compare each element of the slices
-		for i := range wb.left {
-			if wb.left[i].Name().String() != newConf.Left[i] {
-				// Resetting the left motor list
-				wb.left = make([]motor.Motor, 0)
-
-				for _, name := range newConf.Left {
-					m, err := motor.FromDependencies(deps, name)
-					if err != nil {
-						return errors.Wrapf(err, "no left motor named (%s)", name)
+		} else {
+			// Compare each element of the slices
+			for i := range *curr {
+				select {
+				case <-ctx.Done():
+					return newMotors, resource.NewBuildTimeoutError(wb.Name())
+				default:
+				}
+				if (*curr)[i].Name().String() != (*fromConfig)[i] {
+
+					for _, name := range *fromConfig {
+						m, err := motor.FromDependencies(deps, name)
+						if err != nil {
+							return newMotors, errors.Wrapf(err, "no left motor named (%s)", name)
+						}
+						newMotors = append(newMotors, m)
 					}
-					wb.left = append(wb.left, m)
+					break
 				}
-				break
 			}
 		}
+		return newMotors, nil
 	}
 
-	if len(wb.right) != len(newConf.Right) {
-		// Resetting the left motor list
-		wb.right = make([]motor.Motor, 0)
+	//Check if wb.left is different from newConf.Left before changing wb.left
+	left, err := updateMotors(&wb.left, &newConf.Left)
+	wb.left = left
```

Comment:
Note to reviewers: the ordering here (i.e., reassigning `left` and `right` before err checking) is consistent with how the previous code was written. Not sure if that was intentional or not (it stood out to me but not sure if it's actually concerning or not), but wanted to flag it here.

Reply:
So I think `left` or `right` will never get set to nil, though they may get set to an empty list in the case where an error occurs while processing the first element of `fromConfig`. The previous code updated `left` and `right` eagerly; by always returning the `newMotors` list from `updateMotors()` and setting before error checking, we do a similarly eager update.

Reply:
Ah I see. Seems fine to keep the old pattern.

Reply:
Yeah they're never nil, they're an empty list.